metasynth/distribution/regex.py

```diff
@@ -0,0 +1,94 @@
+"""Distribution for structured strings, using regexes."""
+from __future__ import annotations
+
+from typing import Union, Optional
+
+from regexmodel import RegexModel, NotFittedError
+
+from metasynth.distribution.base import metadist, BaseDistribution, UniqueDistributionMixin
+
+
+@metadist(implements="core.regex", var_type="string", version="2.0")
+class RegexDistribution(BaseDistribution):
+    """Distribution for structured strings.
+
+    Examples that this distribution should work reasonably for are: email,
+    ID's, telephone numbers, ip addresses, etc.
+
+    Valid regexes are ([a][b]|[c][d]), or [A-Z][0-9]{2,3}. During the fit, the probabilities
+    of each of the branches in the regex is also recorded so that synthetic values can be
+    drawn from a better distribution.
```

Comment:
Suggestion: make this clearer and separate it from the data generation comment. For example:

Allowed regex elements are
- parentheses ()
- square brackets []
- single characters (e.g., a, \n, 3)
- the ranges [A-Z], [a-z], [0-9] (but not [B-F] or [4-7], then use [4567])
- repetition specification with two numbers {3,6} but not one {6}

For more information on the allowed regex subset, see the regexmodel package.